### PR TITLE
New check for checking HTTP responses for each instances on a ELB

### PR DESCRIPTION
**Pull request by markwallsgrove**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass

#### New Plugins

- [ ] Tests

- [x] Add the plugin to the README

- [x] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Purpose
Check a HTTP endpoint for a HTTP200 for each instance attached to a ELB. I plan on using a separate endpoint for non critical health issues.

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for the contribution! I will review it shortly.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ This CHANGELOG follows the format listed at [Keep A Changelog](http://keepachang
 
 ## [Unreleased]
 
+## [8.4.0] - 2017-09-29
```
Comment (majormoses) [inline CHANGELOG.md]: Please leave the version bumping and dating to the maintainers. All changes should stay under `## [Unreleased]` until a maintainer is ready to create a release. https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md#unreleased-change

File: bin/check-elb-instances-endpoints.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+#    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#    SOFTWARE.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk-v1'
```
Comment (majormoses) [inline bin/check-elb-instances-endpoints.rb]: with this being a new check we should write this against the v2 api. Eventually we will be going through and updating the checks for v2 and dropping v1 support due to version conflicts with newer versions of some gems (json if I recall correctly) with newer sensu setups.

File: bin/check-elb-instances-endpoints.rb [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+# check each node by calling a http endpoint
+class HTTPCheckELBNodes < Sensu::Plugin::Check::CLI
+  option :aws_access_key,
```
Comment (majormoses) [inline bin/check-elb-instances-endpoints.rb]: We are trending towards not passing the key on the CLI at all due to the number of people not properly using token substitution and lack of redaction. If you use the v2 and: https://github.com/sensu-plugins/sensu-plugins-aws/blob/8.3.0/lib/sensu-plugins-aws/common.rb you will get env and IAM which are the two methods we are going to use going forward.

File: bin/check-elb-instances-endpoints.rb [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+         short: '-a AWS_ACCESS_KEY',
+         long: '--aws-access-key AWS_ACCESS_KEY',
+         description: "AWS Access Key. or use ENV['AWS_ACCESS_KEY']",
+         default: ENV['AWS_ACCESS_KEY']
+
+  option :aws_secret_access_key,
```
Comment (majormoses) [inline bin/check-elb-instances-endpoints.rb]: see above

File: bin/check-elb-instances-endpoints.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+         description: 'The name of the ELB',
+         required: true
+
+  option :path,
+         short: '-p PATH',
+         long: '--path PATH',
+         description: 'path to check',
+         required: true
+
+  option :use_https,
```
Comment (majormoses) [inline bin/check-elb-instances-endpoints.rb]: I'd prefer to default to a secure setup and have people override when they want to be insecure. I'd also accept just exposing an option such as:
```ruby
    option :protocol,
           long: '--protocol PROTO',
           in: %(http https),
           default: 'https'
```

File: bin/check-elb-instances-endpoints.rb (hunk 2 above)
Comment (runningman84) [reply bin/check-elb-instances-endpoints.rb]: yes this is a really big issue which prevents us from upgrading to 1.x

Comment (majormoses) [reply bin/check-elb-instances-endpoints.rb]: Can we re-write this to be against v2?

Comment (markwallsgrove) [reply bin/check-elb-instances-endpoints.rb]: :+1:

**Comment by majormoses** [comment]
Just checking if you still planned on re-writing it against sdk v2 as we are in the process of re-writing existing ones and will not accept a new check with v1. You can read the meta issue here: https://github.com/sensu-plugins/sensu-plugins-aws/issues/240

**Comment by markwallsgrove** [comment]
@majormoses yep, I would like to.

**Comment by majormoses** [comment]
Just a friendly reminder this is still open and if possibly I'd like to get this across the finish line but it's in your court.

**Comment by markwallsgrove** [comment]
:+1:

**Pull request closed without merging**